handler/src/main/java/io/netty/handler/ssl/SniHandler.java

```diff
@@ -100,7 +100,12 @@ public SslContext sslContext() {
     protected final void onLookupComplete(ChannelHandlerContext ctx,
                                           String hostname, Future<SslContext> future) throws Exception {
         if (!future.isSuccess()) {
-            throw new DecoderException("failed to get the SslContext for " + hostname, future.cause());
+            final Throwable cause = future.cause();
+            if (cause instanceof Error) {
+                throw (Error) cause;
+            } else {
```

Comment:
nit: you can also remove the else here.

Reply:
@normanmaurer I made this change, squashed, and force pushed here.